File: backend_go/internal/models/income_category.go
package models

import "time"

// IncomeCategory represents an income category scoped to a household.
// Sequelize model: IncomeCategory — table: income_categories
type IncomeCategory struct {
	ID          uint      `gorm:"primaryKey;autoIncrement" json:"id"`
	Name        string    `gorm:"type:varchar(255);not null" json:"name"`
	NameHe      *string   `gorm:"type:varchar(255)" json:"nameHe"`
	Icon        *string   `gorm:"type:varchar(100)" json:"icon"`
	Color       *string   `gorm:"type:varchar(50)" json:"color"`
	SortOrder   int       `gorm:"default:0" json:"sortOrder"`
	HouseholdID uint      `gorm:"not null;index" json:"householdId"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`

	Household Household `gorm:"foreignKey:HouseholdID" json:"-"`
}

func (IncomeCategory) TableName() string { return "income_categories" }

File: models/incomeCategory.js
module.exports = (sequelize, DataTypes) => {
  const IncomeCategory = sequelize.define('IncomeCategory', {
    id: {
      type: DataTypes.INTEGER,
      primaryKey: true,
      autoIncrement: true
    },
    name: {
      type: DataTypes.STRING(100),
      allowNull: false
    },
    nameHe: {
      type: DataTypes.STRING(100),
      allowNull: true
    },
    icon: {
      type: DataTypes.STRING(100),
      allowNull: false
    },
    color: {
      type: DataTypes.STRING(20),
      defaultValue: '#607D8B'
    },
    sortOrder: {
      type: DataTypes.INTEGER,
      defaultValue: 0
    },
    householdId: {
      type: DataTypes.INTEGER,
      allowNull: false,
      references: {
        model: 'households',
        key: 'id'
      }
    },
    isActive: {
      type: DataTypes.BOOLEAN,
      defaultValue: true
    },
    parentCategoryId: {
      type: DataTypes.INTEGER,
      allowNull: true,
      defaultValue: null
    }
  }, {
    timestamps: true,
    tableName: 'income_categories'
  });

  return IncomeCategory;
};
